server/controllers/BookmarkController.js

```diff
@@ -0,0 +1,140 @@
+import db from '../database/models/index';
+
+const { Bookmark } = db;
+
+/**
+ * @param {param} req
+ * @param {param} res
+ * @param {func} next
+ */
+class BookmarkController {
+  /**
+ * @param {obj} req
+ * @param {obj} res
+ * @param {obj} next
+ * @returns {json} bookmark article
+ */
+  static bookmarkArticle(req, res, next) {
+    const { articleSlug } = req.body;
+
+    const { userId } = req.decoded;
+
+    if (!articleSlug) {
+      return res.status(400).json({
+        message: 'Article slug not provided'
+      });
+    }
+
+    Bookmark.findOne({
+      where: {
+        articleSlug,
+        userId
+      }
+    })
+      .then((bookmark) => {
+        if (bookmark) {
+          res.status(400).json({
+            message: 'You have already bookmarked this article'
+          });
+        } else {
+          Bookmark.create({
+            articleSlug,
+            userId
+          })
+            .then((data) => {
+              res.status(201).json({
+                message: 'Article bookmarked successfuly',
+                data
+              });
+            })
+            .catch(next);
+        }
+      });
+  }
+
+  /**
+ * @param {obj} req
+ * @param {obj} res
+ * @param {obj} next
+ * @returns {json} get list of bookmarks
+ */
+  static getBookmarks(req, res, next) {
+    Bookmark.findAll({
+      where: {
+        userId: req.decoded.userId
+      }
+    })
+      .then((bookmark) => {
+        if (!bookmark) {
+          return res.status(400).json({
+            message: 'You have not bookmarked any article'
+          });
+        }
+        return res.status(200).json({
+          bookmark
+        });
+      })
+      .catch(next);
+  }
+
+  /**
+ * @param {obj} req
+ * @param {obj} res
+ * @param {obj} next
+ * @returns {json} articles
+ */
+  static getSingleBookmark(req, res, next) {
+    const { userId } = req.decoded;
+    Bookmark.findOne(
+      {
+        where: {
+          articleSlug: req.params.slug,
+          userId
+        }
+      }
+    )
+      .then((bookmark) => {
+        if (!bookmark) {
+          return res.status(404).json({
+            message: 'You did not bookmark this article'
+          });
+        }
+        return res.status(200).json({
+          bookmark
+        });
+      })
+      .catch(next);
+  }
+
+  /**
+ * @param {obj} req
+ * @param {obj} res
+ * @param {obj} next
+ * @returns {json} articles
+ */
+  static deleteBookmark(req, res, next) {
+    Bookmark.findOne(
```

Comment:
I see you used this same line of code in a couple of places,(searching for bookmarks using the article slug and userId). I believe it will help you uphold the (DRY) principle if you can write it as a middleware function instead and reuse it in all the places where you need it.